### DISCUSSION TITLE
Scaling beyond one replica

### DISCUSSION THREAD

**Discussion by dparker18** [jertel/elastalert2#938] [Q&A] [answered]
I see that there are issues running multiple elastalert replicas with the same rules on the same data (see https://github.com/jertel/elastalert2/discussions/451 and https://github.com/Yelp/elastalert/issues/478)

I'm looking to do this not for HA, but for scale. Is there any harm in running 2 elastalerts using the same writeback indices if they both have different set of rules that don't overlap? Or is there a better way?

**Comment by jertel** [comment, marked as answer]
There could be some collision of data since a second instance would be querying the same elastalert indices to determine current state. For example, it would see unsent/failed/pending alerts from the other instance, and potentially attempt to act on that data. 

To eliminate any data collisions such as this, ensure each instance is using a different set of elastalert_* indices. You can do this via the `writeback_index` setting.

**Reply by dparker18** [reply]
Makes sense. Thank you!
